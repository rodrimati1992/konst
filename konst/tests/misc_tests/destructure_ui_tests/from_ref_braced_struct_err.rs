struct Braced {
    bar: u8,
    baz: u8,
    qux: u8,
}


const fn to_bar(foo: &Braced) -> impl Sized {
    konst::destructure!{Braced{bar, baz, qux} = foo}

    (foo, bar, baz)
}

const fn to_bar2(foo: &Braced) -> impl Sized {
    konst::destructure!{Braced {bar, baz, qux}: &Braced = foo}

    (foo, bar, baz)
}

const fn to_bar_type_as_pat(foo: &Braced) -> impl Sized {
    konst::destructure!{Braced<> {bar, baz, qux}: &Braced = foo}

    (foo, bar, baz)
}

fn main(){}