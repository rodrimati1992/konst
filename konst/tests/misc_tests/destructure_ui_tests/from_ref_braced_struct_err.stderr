error[E0308]: mismatched types
 --> tests/misc_tests/destructure_ui_tests/from_ref_braced_struct_err.rs:9:5
  |
9 |     konst::destructure!{Braced{bar, baz, qux} = foo}
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |     |
  |     expected `Braced`, found `&Braced`
  |     arguments to this function are incorrect
  |
note: function defined here
 --> src/macros/destructuring.rs
  |
  | pub const fn assert_same_type<T>(this: T, that: T) {
  |              ^^^^^^^^^^^^^^^^
  = note: this error originates in the macro `$crate::__destructuring__type_assert` which comes from the expansion of the macro `konst::destructure` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/destructure_ui_tests/from_ref_braced_struct_err.rs:15:5
   |
15 |     konst::destructure!{Braced {bar, baz, qux}: &Braced = foo}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |     |
   |     expected `Braced`, found `&Braced`
   |     arguments to this function are incorrect
   |
note: function defined here
  --> src/macros/destructuring.rs
   |
   | pub const fn assert_same_type<T>(this: T, that: T) {
   |              ^^^^^^^^^^^^^^^^
   = note: this error originates in the macro `$crate::__destructuring__type_assert` which comes from the expansion of the macro `konst::destructure` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/misc_tests/destructure_ui_tests/from_ref_braced_struct_err.rs:21:5
   |
21 |     konst::destructure!{Braced<> {bar, baz, qux}: &Braced = foo}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |     |
   |     expected `Braced`, found `&Braced`
   |     expected due to this
   |
   = note: this error originates in the macro `$crate::__destructuring__type_assert` which comes from the expansion of the macro `konst::destructure` (in Nightly builds, run with -Z macro-backtrace for more info)
